### ISSUE TITLE
Cannot update complex item with pn.depends

### ISSUE THREAD

**Issue by sg2475962** [holoviz/panel#4950]
environment: panel 1.0.3 (minimal install, python 3.11.3)

#### Description of expected behavior and the observed behavior
Expected behaviour: show_value_fails should update when minimal changes state
Observed behaviour: show_value_fails never changes, and outputs an error (seen in stacktrace at the bottom)

```
import panel as pn

minimal = pn.widgets.Select(name='test', options=['1','2','3'],value='1')
@pn.depends(minimal)
def show_value_works(value):
    return pn.panel(f'this works: {value}')

@pn.depends(minimal)
def show_value_fails(value):
    return pn.Row(pn.panel(value),pn.panel(' this does not work?'))

layout = pn.Column(minimal,
                   show_value_works,
                   show_value_fails)
layout.servable()
```

stacktrace:
<details>
<summary>Command line stacktrace</summary>

```

AttributeError: 'str' object has no attribute 'param'
2023-05-26 16:07:28,225 Exception in callback functools.partial(<bound method IOLoop._discard_future_result of <tornado.platform.asyncio.AsyncIOMainLoop object at 0x7feb5560e790>>, <Task finished name='Task-783' coro=<ServerSession.with_document_locked() done, defined at /home/it-499/mambaforge/envs/panel/lib/python3.11/site-packages/bokeh/server/session.py:77> exception=AttributeError("'str' object has no attribute 'param'")>)
Traceback (most recent call last):
  File "/home/it-499/mambaforge/envs/panel/lib/python3.11/site-packages/tornado/ioloop.py", line 738, in _run_callback
    ret = callback()
          ^^^^^^^^^^
  File "/home/it-499/mambaforge/envs/panel/lib/python3.11/site-packages/tornado/ioloop.py", line 762, in _discard_future_result
    future.result()
  File "/home/it-499/mambaforge/envs/panel/lib/python3.11/site-packages/bokeh/server/session.py", line 98, in _needs_document_lock_wrapper
    result = await result
             ^^^^^^^^^^^^
  File "/home/it-499/mambaforge/envs/panel/lib/python3.11/site-packages/panel/reactive.py", line 424, in _change_coroutine
    state._handle_exception(e)
  File "/home/it-499/mambaforge/envs/panel/lib/python3.11/site-packages/panel/io/state.py", line 432, in _handle_exception
    raise exception
  File "/home/it-499/mambaforge/envs/panel/lib/python3.11/site-packages/panel/reactive.py", line 422, in _change_coroutine
    self._change_event(doc)
  File "/home/it-499/mambaforge/envs/panel/lib/python3.11/site-packages/panel/reactive.py", line 440, in _change_event
    self._process_events(events)
  File "/home/it-499/mambaforge/envs/panel/lib/python3.11/site-packages/panel/reactive.py", line 379, in _process_events
    self.param.update(**self_events)
  File "/home/it-499/mambaforge/envs/panel/lib/python3.11/site-packages/param/parameterized.py", line 1902, in update
    self_._batch_call_watchers()
  File "/home/it-499/mambaforge/envs/panel/lib/python3.11/site-packages/param/parameterized.py", line 2063, in _batch_call_watchers
    self_._execute_watcher(watcher, events)
  File "/home/it-499/mambaforge/envs/panel/lib/python3.11/site-packages/param/parameterized.py", line 2025, in _execute_watcher
    watcher.fn(*args, **kwargs)
  File "/home/it-499/mambaforge/envs/panel/lib/python3.11/site-packages/panel/param.py", line 827, in _replace_pane
    self._update_inner(new_object)
  File "/home/it-499/mambaforge/envs/panel/lib/python3.11/site-packages/panel/pane/base.py", line 700, in _update_inner
    new_pane, internal = self._update_from_object(
                         ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/it-499/mambaforge/envs/panel/lib/python3.11/site-packages/panel/pane/base.py", line 655, in _update_from_object
    cls._recursive_update(old_object, i, old, new)
  File "/home/it-499/mambaforge/envs/panel/lib/python3.11/site-packages/panel/pane/base.py", line 629, in _recursive_update
    old.param.update(**new_params)
    ^^^^^^^^^
AttributeError: 'str' object has no attribute 'param'
```

</details>
No errors in js console.

command run:
`panel serve --port=28283 --allow-websocket-origin=* warming_level.py --autoreload --admin`

**Comment by Hoxbro** [comment]
Thank you for the bug report. Luckily this is already fixed with https://github.com/holoviz/panel/pull/4946.

The fix will be out in the next release, 1.0.4.

**Comment by sg2475962** [comment]
Thanks so much. Glad it's fixed!
